### DISCUSSION TITLE
Release Automation

### DISCUSSION THREAD

**Discussion by Incpi** [dbeck121/CPI-Helper-Chrome-Extension#217] [Ideas]
Hello @dbeck121 ,

I've the same automation for my Dark CPI extension (thought to adding here).

- This workflow creates a GitHub release based on the version extracted from 'manifest.json'.
- The zip file is located in the 'bin' directory, renamed according to the naming convention, and attached to the release.
- Version extracted from 'manifest.json' will be used for both the tag and release name.

To put it briefly, when you update the bin folder in the main branch using a zip file, the release will attach the same zip file and extract the data metadata directly from the file.

No need to add "Github token" it will taken care by action.

Let's have a discussion or thoughts. BTW, #216 <- PR I Closed temporarily because merge won't result in Action.

Thanks

**Comment by dbeck121** [comment]
@incpi that is sth I was thinking about the last days so it is really great that you come up with this.

We are also using manifest v2 for firefox right now. So we could adjust it to release manifest v2 and 3 file. Or try to solve manifest v3 on firefox but it seems like the background runner is not supported in FF.

So let's go the way with the github action!

**Comment by Incpi** [comment]
Hi @dbeck121,

Should I enable action? 

naming convection I follow

- release: Extension @<version like 0.0.0>
- tag name : version_<version like 0.0.0>

We already follow below pattern, So same will be applied to this action.

- **Format**: `Major.Minor.Maintenance.Build` 
  - **Build** (optional): Used for internal purposes such as build numbers or source control revisions. This number will not appear in the changelog and is only included in maintenance versions.


surely comes with limitation,

release won't be updates if version is in patch instead original version will be updated "1.3.5" as latest then next version 1.3.5.1 will be merge in 1.3.5 ,

File name we already have process in place. So no worry for that.

**Reply by dbeck121** [reply]
I added 2 comments to push.

So from my side lets talk about these comments and enable.

**Reply by dbeck121** [reply]
Maybe step 2 could be that we create zips also with github actions :-)

**Reply by Incpi** [reply]
Didnot follow what you said,

I got zip along with workflow (new but i can explore if we can, let me check docs..)
what is 2nd comments?

**Reply by dbeck121** [reply]
> Didnot follow what you said,
> 
> I got zip along with workflow (new but i can explore if we can, let me check docs..) what is 2nd comments?

2nd comment is an idea of mine. Check https://github.com/dbeck121/CPI-Helper-Chrome-Extension/blob/devops/.github/workflows/create_bin_files.yml

This should create the zip files when pushing to dev automatically. Your PR then creates the release!

I am not sure if my github action works. How do you debug it without really push it? :-)

**Reply by Incpi** [reply]
Hello @dbeck121

We have same thought..😄

I have dummy private copy of this plugin, On windows not a great support of zip and bash, So, bash on VM (Ubuntu) then github with vscode.

I know there are tools but i don't know. i recently learn for my plugin.

___
I mix yours and mine.

- Completed in #216 || lost track of time today working on last 6 hours ... 
- I would like to suggest to don't deploy on dev instead do on main as default branch.
- In dev we made too many changes, to have this workflow.
- Perquisites: Delete bin folder content for initial version or later from release, then it will work as expected. I am not deleting from git automation ( workflow only updates )
